Added data-rellax-percentage.

p-realinho opened the pull request:
Added the attribute `data-rellax-percentage`, to make it possible to easily offset elements (relatively to their initial position).
This avoids elements further down the page to be too far away from their initial position when you scroll past them.
For example, if you set the `data-rellax-percentage` to `0.5`, the element will be in it's initial position when it's scrolled into the viewport's center.

moeamaya commented:
I think this is an awesome concept, but I'm running into bugs when testing it on `demo.html`, with double scrollbars and the top elements reacting strangely. 

I added `data-rellax-percentage="0.5"` to the element on Line 267 with the class `md-facebook`. 

![rellax-optim](https://cloud.githubusercontent.com/assets/1373293/18188278/c625ca86-7065-11e6-9e5e-ff4e9d8f2acb.gif)

p-realinho commented:
Thanks for your response! 

I did some more testing, fixed one other related thing, and added the test page I was using.

The issue you were having with the demo page, was because since you added the percentage attribute, that element had to initialize further down than it used to, in order to be in it's original position when scrolled into the viewport's center.
So it was causing the `<main>` element to have a scrollbar.
I added an `overflow: hidden` to the `.main`, and that fixed it.

moeamaya commented:
Looking good! The only other change I would make is that if you use percentage the speed ranges is limited -5 to 5, otherwise it's a bit buggy

p-realinho commented:
Hi! Sorry for the delayed response ... 
So, I was just looking through the code to figure out the best way and place to implement the -5/5 limit, and I noticed that the current "speed limit" implementation doesn't have any effect, if the user is using the `data-rellax-speed` attr.
You do limit the speed, but only if it's set in the options, when the `rellax` object is created.

Now, if that checks out, my questions are:
Do you want me to fix that and implement both speed limits, maybe give the percentage a nicer name, make it possible to set it on the "global" options, etc ... ? 
**OR**
Do you guys maybe wanna do that, cause this is actually my first PR and I'm not toooo experienced with Js so I'm kind of afraid to make a big useless mess? :D
**OR**
Would you maybe consider dropping the speed limits, and instead, have something like a "Recommended" section on the readme/docs, so users know what kind of weird behavior to expect when setting the speed to higher/lower values, and how to fix those bugs?

With that said, dropping the limits would probably make more sense (imo), cause it gives more flexibility for the user to "go crazy", and it's less code to write (in the spirit of keeping it lightweight).

moeamaya commented:
Yea the speed option was kinda a placeholder for more options in the future. It's pretty useless right now. I thought about maybe adding a random so it wasn't always the same, but meh maybe later.

Yea I like the idea of a global percentage option, that way you would only have to individually set speed on each rellax element. `fit: true` or `frame: true` or something??

For now I'll keep the limits, but should definitely add something to README.

No worries about messing things up. It's 100 line file so pretty easy to have a good mental model of the whole thing. Plus this is our first big repo that we've been handling PRs with, so we're all learning 👯

moeamaya commented:
Pulled down the most recent version and seems like the demo.html doesn't appear to allow elements at different speeds? Just checking if you're still in the process?

p-realinho commented:
Yup. Still on it.
I haven't had much time lately :/


On 9/10/2016 00:40, Moe Amaya wrote:
>
> Pulled down the most recent version and seems like the demo.html 
> doesn't appear to allow elements at different speeds? Just checking if 
> you're still in the process?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/dixonandmoe/rellax/pull/13#issuecomment-246069389>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AJky2_xQW4cyId_ofMqc-taJb6yxewVpks5qoe5bgaJpZM4JzC5_>.
>

-- 
Cumprimentos,
Pedro Realinho.

moeamaya commented:
Seriously, no rush. Just checking in

p-realinho commented:
Okay, so, I fixed that last problem you mentioned.
I created it the last time I tried to fix inline styles not being applied. 
Your implementation only worked with transforms. When I changed it, it was working with everything else, but not with transforms.
Now, it stores the element's inline styles, checks if there's a transform somewhere in there, and extracts it.
Later on, when updating the element's position, it appends those initial transforms to the translate3d, and prepends the other inline styles.

I'll implement the `frame` option now.

Let me now if there's anything else, that I might have not noticed, waiting to be fixed. :)

p-realinho commented:
I was using this on a project for a client and found a problem with using inline `transition: transform` I think. 
Just posting here so I don't forget, I'll fix it later.

p-realinho commented:
Hi! I've been using this for some time now and haven't run into any issue.
Is there anything else you want me to change or add?

moeamaya commented:
Hey Pedro, wasn't sure where you were at with this, but I'll take a look this weekend and let you know if I find any issues then we can proceed.

moeamaya commented:
Hi Pedro, I promise I haven't forgotten this but it's been an exceptionally busy time at the office and with my moving to another city. Give me a couple more weeks to get to this.

p-realinho commented:
Sure thing, no worries. Good luck with all of that!

moeamaya commented:
Hey @p-realinho, just merged in your PR 🎉
Only 4 months per pull request :)

I need to add in some docs and a few other things, but I'm ready to get this to v1.0 by the end of the year. 12 days left to get it done!

p-realinho commented:
I just saw it! Thanks for the shout out in the readme too! ahaha
If there's anything else I can help with, feel free to hmu :)
